tests/webTables.spec.ts

```diff
@@ -1,222 +1,133 @@
+        /* 2. Locate the owner by the phone number "6085552765". Extract the Pet name displayed in the table for the owner and save it to the variable. Click on this owner. */
+        const phoneNumber = '6085552765'
+        
+        // Capture the row with the above associated phone number
+        const row = page.getByRole('row', {name: phoneNumber})
+        
+        // Extract the pet name from the row
+        const petName = await row.locator('td').nth(4).textContent()!
+    
+        // Click on the owner to go to the owner information page
+        await row.getByRole('link').click()
+    
+        /* 3. On the Owner Information page, add the assertion that "Telephone" value in the Onner Information card is "6085552765" */
+        // Validate that the phone number is listed correctly
+        await expect(page.getByRole('row', {name: 'Telephone'}).getByRole('cell').last()).toHaveText(phoneNumber)
+    
+        /* 4. Add the assertion that Pet Name in the Owner Information card matches the name extracted from the page on the step 2 */
+        // Validate the correct pet name by finding 'name' in the first pet information table
+        // Assumes we are searching for one pet
+        await expect(page.locator('dt', {hasText: 'Name'}).locator('..').locator('dd').first()).toHaveText(petName)
+    });
+    
+    test('Validate pets of the Madison city', async ({ page }) => {
+        // Wait for the table and its data to load
+        await page.waitForSelector('table')
     
-    // Collect all rows in the table
-    const allRows = await page.getByRole('row').all()
-
-    // Iterate through each row of the table
-    for (const row of allRows) {
-        // Collect each cell in the row
-        const rowData = row.getByRole('cell')
-
-        // Make sure the row has cells
-        if((await rowData.all()).length > 0){
-            if (await rowData.nth(2).textContent() === 'Madison') {
-                // Locate the pet name in the row
-                // Trim the result so there are no extra white spaces around the pet name
-                actualPetsList.push((await rowData.nth(4).textContent())?.trim())
+        // Create a list of the expected Madison pets
+        const expectedPetsList = ['Freddy', 'George', 'Leo', 'Mulligan']
+    
+        // Collect rows in the table referring to the city of Madison (mitigating the potential for an owner named Madison)
+        const madisonRows = await page.getByRole('row', {name: 'Madison'}).all()
+        
+        /* 2. On the Owners page, perform the assertion that Madison city has a list of pets: Leo, George, Mulligan, Freddy */
+        // Create an empty list for what we actually find
+        let actualPetsList: string[] = []
+    
+        // Iterate through the rows to extract the pet names
+        for(const row of madisonRows){
+            // Ensure we're not dealing with an empty row
+            if((await row.getByRole('cell').all()).length > 0){
```

Comment:
Same here... it's not needed to ensure that you don't have empty cells. This is over-engineering.

Reply:
@b-w-cole it seems like you have missed this fixed in your latest updates